### PR TITLE
feat(lvol): snapshots enumeration method for lvol store

### PR DESCRIPTION
**Pull request by mtzaurus**
Lvol object now has a dedicated method for listing all snapshots that reside on the pool.

### CODE REVIEW

**Comment by mtzaurus** [comment]
bors try

File: io-engine/src/lvs/lvs_store.rs [hunk 1]
```diff
@@ -644,6 +645,35 @@ impl Lvs {
         Ok(())
     }
 
+    /// return an iterator for enumerating all snapshots that reside on the pool
+    pub fn snapshots(
+        &self,
+    ) -> Option<impl Iterator<Item = VolumeSnapshotDescriptor>> {
+        if let Some(bdev) = UntypedBdev::bdev_first() {
```
Comment (tiagolobocastro) [inline io-engine/src/lvs/lvs_store.rs:652]: Is this pattern common for all snapshot listing? As in get all the bdevs, then do some filtering and also check if they're a snapshot?
If so could we have a common function that does it which just receives a closure for the initial filter?

**Review by tiagolobocastro — APPROVED** [review]

**Review by hrudaya21 — APPROVED** [review]

**Comment by mtzaurus** [comment]
bors try

**Comment by mtzaurus** [comment]
bors cancel

**Comment by mtzaurus** [comment]
bors try

**Comment by mtzaurus** [comment]
borse merge

**Comment by mtzaurus** [comment]
bors merge

**Comment by mtzaurus** [comment]
bors merge

**Pull request merged**